### PR TITLE
Infinite Loading Inbox

### LINKED ISSUE #1432: Infinite inbox loading
Right now it only loads one page. In v1 this loaded infinitely, so it would be nice to get this into 2.0 - that said, I'm also OK with leaving it out if we get pressed for time.

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read CONTRIBUTING.md
- [x] I have described what this PR contains
- [x] If this PR alters the UI, I have attached pictures/videos
- [x] This PR addresses one or more open issues that were assigned to me:
      - closes #1432 

# Pull Request Information

Frontend changes for https://github.com/mlemgroup/MlemMiddleware/pull/74.

Mostly just updates to use `InboxFeedLoader` and friends. The main notable change is that items are not all hidden when "mark all as read" is used. This is consistent with how the feed behaves when read items are filtered out and the user marks an item as read (i.e., it remains visible)

This PR also includes a minor tweak to show the media type indicator when developer mode is active, rather than guarding it in `#DEBUG`. This should make daily driving a local dev build nicer, plus offer that insight in the production app as well if needed.

### CODE REVIEW

**Comment by Sjmarf** [comment]
The inbox doesn't respect the "Infinite scroll" setting - is this intentional?  Also, the "Load more" button flashes briefly when loading a new page, which isn't ideal.

This PR closes #1479 💯

**Review by Sjmarf — CHANGES_REQUESTED** [review]
Found a few bugs:
- The "inbox is outdated" popup isn't shown anymore when the tab badge count changes.
- The "Load more" button flashes briefly when loading a new page.
- The inbox doesn't respect the "Infinite scroll" setting - is this intentional? 

This PR closes #1479 💯

**Comment by EricBAndrews** [comment]
> The inbox doesn't respect the "Infinite scroll" setting - is this intentional?

Yes. The theory is that manual page loading is a time management tool, but since people don't really "scroll" their inbox like they do a feed it doesn't make sense in that context.

**Comment by EricBAndrews** [comment]
> The "Load more" button flashes briefly when loading a new page.

This one was a middleware issue--fixed [here](https://github.com/mlemgroup/MlemMiddleware/pull/74/commits/a57a794b3b3678073fc70416785184745d697aaa). It was a super trivial change (just needed to move the loading state assignment to after the item assignment), so I went ahead and merged the middleware side--if there are other issues that need a middleware fix I'll open a new one

File: Mlem/App/Views/Shared/EndOfFeedView.swift [hunk 1]
```diff
@@ -45,6 +46,12 @@ struct EndOfFeedView: View {
                         loadMore()
                     }
                     .buttonStyle(.bordered)
+                } else {
+                    if developerMode {
+                        Text("IDLE")
```
Comment (Sjmarf) [inline Mlem/App/Views/Shared/EndOfFeedView.swift]:
```suggestion
                        Text(verbatim: "IDLE")
```

File: Mlem/Localizable.xcstrings [hunk 2]
```diff
@@ -697,6 +697,9 @@
     },
     "I think I've found the bottom!" : {
 
+    },
+    "IDLE" : {
+
```
Comment (Sjmarf) [inline Mlem/Localizable.xcstrings]:
```suggestion
```

**Review by Sjmarf — APPROVED** [review]
LGTM! 👍

**Pull request merged**